### PR TITLE
feat: add configurable file open behavior for external launches

### PR DESCRIPTION
**Pull request by lambdalisue**
## Summary

- Add `FileOpenBehavior` config enum with three routing strategies: `new_window`, `last_focused`, and `current_screen`
- Implement intelligent window selection logic in IPC module to route external file/directory opens based on user preference
- Add General tab in Preferences UI to configure file open behavior
- Refactor window management helpers to support generic window focusing

CleanShot 2026-02-12 at 02 35 05

## Why

When users open files from Finder, CLI, or other external sources, Arto previously always created new windows. This could lead to window proliferation and inconsistent behavior. Users needed control over how external opens are routed:

- **New Window**: Always create a new window (previous behavior)
- **Last Focused**: Open in the last focused visible window (reduces window clutter)
- **Current Screen**: Open in a window on the cursor's current screen (multi-monitor workflow)

The default is `last_focused` to provide a more streamlined experience while maintaining the ability to spawn new windows when desired.

## Implementation

The implementation is structured in three logical layers:

### 1. Configuration Type (`9c2860f`)

**Files:** `config/app_config/file_open_behavior.rs`, `config/app_config.rs`

- Define `FileOpenBehavior` enum with three variants
- Serialize as snake_case JSON (`"new_window"`, `"last_focused"`, `"current_screen"`)
- Store in `config.json` under `fileOpen` key with default `LastFocused`
- Include roundtrip serialization tests

### 2. IPC Routing Logic (`ec274cd`)

**Files:** `ipc.rs`, `window/main.rs`, `window.rs`

- **`choose_open_target()`**: Pure function selecting target window based on behavior and window state
- **`collect_visible_window_selection_inputs()`**: Gathers window positions and checks screen overlap
- **`is_window_on_display()`**: Detects if a window overlaps with display bounds using geometry intersection
- **Behavior wrappers**: `open_file_with_behavior()`, `open_directory_with_behavior()`, `reopen_with_behavior()`
- Opens file/directory in selected window via `AppState` API, or creates new window as fallback
- Comprehensive unit tests covering all routing scenarios and edge cases

**Window management refactoring:**
- Remove MainApp-specific helpers (`is_main_app_window_visible`, `show_main_app_window`, `get_main_app_window`)
- Add generic `focus_window(WindowId)` and `show_and_focus_hidden_window()`
- Use existing `WINDOW_STATES` mapping to access any window's state

### 3. Preferences UI (`4a3b72f`)

**Files:** `preferences_view/tabs/general_tab.rs`, `preferences_view/tabs.rs`, `preferences_view/main_view.rs`

- New General tab component with option cards for file open behavior selection
- Reorder navigation tabs for better logical grouping: General → Theme → Directory → Window Size → Window Position → Sidebars → About
- Real-time config updates via `CONFIG` global state

## Test Plan

**Unit Tests (Completed):**
- [x] `FileOpenBehavior` serialization roundtrip
- [x] `choose_open_target()` logic covering all behaviors and edge cases
- [x] `is_window_on_display()` overlap detection with various window/display geometries

**Manual Testing (Pending):**
- [ ] Open file from Finder with `new_window` behavior → New window created
- [ ] Open file from Finder with `last_focused` behavior → Opens in last focused window
- [ ] Open file from Finder with `current_screen` behavior → Opens in window on cursor screen
- [ ] Multi-monitor setup with `current_screen` behavior → Correctly detects cursor screen
- [ ] Preferences UI saves and persists file open behavior setting
- [ ] No visible windows + `last_focused`/`current_screen` → Falls back to new window creation

### CODE REVIEW

File: desktop/src/ipc.rs [hunk 1]
```diff
@@ -210,33 +212,176 @@ fn handle_event_on_main_thread(
...
+        .map(|ctx| {
+            let metrics = crate::window::metrics::capture_window_metrics(&ctx.window);
+            let is_on_current_screen = current_display_bounds
+                .map(|display| {
+                    is_window_on_display(
+                        display,
+                        LogicalPosition::new(metrics.position.x, metrics.position.y),
+                        LogicalSize::new(metrics.size.width, metrics.size.height),
+                    )
+                })
```
[the remark this answers is not in the document]
Comment (lambdalisue) [reply desktop/src/ipc.rs]: Good catch! You're absolutely right that mixing `outer_position` with `inner_size` creates an inconsistent coordinate space. I'll fix this by using `outer_position()` + `outer_size()` for a consistent outer bounds calculation, which is the correct approach for display overlap detection.

File: desktop/src/ipc.rs [hunk 2]
```diff
@@ -210,33 +212,176 @@ fn handle_event_on_main_thread(
...
+            return;
+        }
+    }
+
+    crate::window::create_main_window_sync(
+        desktop,
+        crate::state::Tab::default(),
+        crate::window::CreateMainWindowConfigParams::default(),
+    );
+}
```
[the remark this answers is not in the document]
Comment (lambdalisue) [reply desktop/src/ipc.rs:333]: You're absolutely correct - this is a regression in the dock click behavior. The current implementation only considers visible windows, so when all windows are hidden, it creates a new window instead of showing the existing hidden MainApp window.

I'll fix this by:
1. Adding a helper to show and focus hidden windows
2. Modifying `reopen_with_behavior()` to first try visible windows, then fall back to showing hidden windows before creating a new one

This will restore the original "dock click shows hidden window" behavior.

File: desktop/src/window/main.rs [hunk 3]
```diff
@@ -201,10 +169,9 @@ pub fn focus_window(window_id: WindowId) -> bool {
 pub fn close_all_main_windows() {
     let windows = list_main_windows();
     windows.iter().for_each(|w| w.close());
-    // Do not clear MAIN_WINDOWS: MainApp uses WindowCloseBehaviour::WindowHides,
-    // so close() hides it instead of destroying it. Clearing would make
-    // get_main_app_window() panic and break IPC window creation.
-    // Dead entries are pruned naturally by register_main_window().
+    // Do not clear MAIN_WINDOWS: with WindowCloseBehaviour::WindowHides,
+    // close() hides windows instead of destroying them. Dead entries are
+    // pruned naturally by register_main_window().
```
[the remark this answers is not in the document]
Comment (lambdalisue) [reply desktop/src/window/main.rs]: Good point - the comment should be more precise about which windows have `WindowCloseBehaviour::WindowHides`. I'll update it to clarify that only the MainApp window is configured with this behavior, while additional windows use the default close behavior.
